src/decoder/mod.rs

```diff
@@ -899,6 +932,16 @@ impl<R: Read + Seek> Decoder<R> {
                 }
                 bytes / 2
             }
+            // TODO: not sure about the following implementation
+            (ColorType::Gray(16), DecodingBuffer::I16(ref mut buffer)) => {
+                reader.read_i16_into(&mut buffer[..bytes / 2])?;
+                if self.photometric_interpretation == PhotometricInterpretation::WhiteIsZero {
+                    for datum in buffer[..bytes / 2].iter_mut() {
+                        *datum = 0xff - *datum
+                    }
+                }
```

Comment:
Let's just report this case as unsupported for now? Unless we have any test file that actually executes the case I wouldn't feel comfortable guessing the semantics.

Reply:
Agreed. 0xff seems like the wrong value to use, but also "white is zero" isn't a very meaningful thing to say about integers that range from -32768 to 32767.

Reply:
Actually looking at the libtiff source code, I think that the variants actually mean "min is white" and "max is white". In which case, the correct conversion would be:
```rust
*datum = -1 - *datum;
```